Tmp ipv6 patch

benjamreis opened the pull request:
Not sure we want to merge this 2 commits, keeping them here for now.

olivierlambert commented on tui/installer/__init__.py:
```diff
@@ -141,6 +141,7 @@ def out_of_order_pool_upgrade_fn(answers):
         results['preserve-settings'] = False
 
     seq = [
+	Step(uis.kernel_warning),
```
Is it expected? (indentation mismatch)

rushikeshjadhav replied:
Thats because I used TAB instead of 8 Spaces. The patch applies fine though. I'll change it during the text update to be same with other lines.

stormi commented:
Looks good to me. I'll modify the text as we devised and fix the indentation at the same time, then merge.

stormi commented:
Merged manually, thanks!

benjamreis committed:
Write `manual_nameservers` and `domain` for IPv6 too

Both fields are used for the 2 IP families and were writen in
an IPv4 specific block.

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Add `NetInterfaceV6` to init an IPv6 interface

Inherits from `NetInterface` to mutualize the code

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
IP mode helpers take IPv6 into account when no IPv4 configured

IPv4 has priority on IPv6 as when both are configured,
the host's management interface's primary address type
will be IPv4

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Use `socket.inet_pton` to validate IP

Add family specific validator as well

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Take IPv6 addresses into account to determine an interface is up

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Enable IPv6 in the host when it is configured

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Write IPv6 conf files for the installer

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Enable manual set of DNS, nameservers and NTP server in autoconf

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Configure IPv6 in TUI

Add a screen to choose to configure IPv4, IPv6 or both
Display the interface screen for IPv4 and/or IPv6

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
Re-enable IPv6 on upgrade when relevant

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>

benjamreis committed:
IPv6 in DebStyle net files

Signed-off-by: BenjiReis <benjamin.reis@vates.fr>
